mysql-test/main/partition.test

```diff
@@ -292,10 +292,11 @@ drop table t1;
 #
 # Bug#36001: Partitions: spelling and using some error messages
 #
---error ER_FEATURE_NOT_SUPPORTED_WITH_PARTITIONING
 CREATE TABLE t1 (a INT, FOREIGN KEY (a) REFERENCES t0 (a))
 ENGINE=MyISAM
 PARTITION BY HASH (a);
+show create table t1;
+drop table t1;
```

Comment:
It is not sufficient to demonstrate that the table can be created. Some functional testing is needed as well, to demonstrate that the foreign keys actually work.

Given that this test uses `ENGINE=MyISAM` and `FOREIGN KEY` constraints are currently implemented only in `ENGINE=InnoDB`, this test seem to demonstrate that things are changing for the worse. We’d allow a DDL operation that was previously rejected because it was known not to work. I don’t think that this change is acceptable.